src/state/pools/index.ts

```diff
@@ -343,6 +344,15 @@ export const PoolsSlice = createSlice({
     },
   },
   extraReducers: (builder) => {
+    builder.addCase(resetUserState, (state) => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      state.data = state.data.map(({ userData, ...pool }) => {
+        return { ...pool }
```

Comment:
Great work ! @memoyil  

One minor thing: Do you think it's dangerous to leave `userData` as undefined ? 

If other places use `pool.userData.xyz` uncarefully, it might throw an error. Right ?

Reply:
True but also the initial state value is undefined, therefore currently all the places that we use, we check its availability beforehand so we are safe

Reply:
Yeah. I check its usages. We are safe for now. Thanks !!! Great work !!!